### DISCUSSION TITLE
Conditionally add values/attributes to response header

### DISCUSSION THREAD

**Discussion by PackiarajSakkananGW** [Kong/kong#8212] [Help]
Hello All,
We have Kong API gate way Infront of monolith app which has both UI & RESTful APIs,
We are using “response-transformer” to add header to the upstream response.
However, we have couple of issues / limitations

1. Unable to add cookie attribute such as HttpOnly;Secure. Also, we need to selectively add SameSite attribute to only one cookie
2. Unable to add “Cache-Control” header only if Content-Type is “text/html”. We need to allow static resources to be cached by browser.

is there a way to achieve above requirement using response-transformer or any other plugin?

Thanks a lot for your time and help

**Comment by kikito** [comment]
Hello,

In order to manage session cookies, you should use the [Session plugin](https://docs.konghq.com/hub/kong-inc/session/).

The Response Transformer plugin is not able to add transformations conditionally, as you noticed. This leaves you two options:

* You could write a [custom plugin](https://docs.konghq.com/gateway/2.7.x/plugin-development/) which adds the transformation like you want.
* You could try the [response-transformer-advanced](https://docs.konghq.com/hub/kong-inc/response-transformer-advanced/) plugins, but that is an Enterprise plugin.

**Reply by PackiarajSakkananGW** [reply]
Thank you for conforming.
"response-transformer-advanded" does not support cookie attribute as well.
Adding/updating cookie attribute is a basic functionality and important from security. at least for migration customer ( from monlith to micro service )

Would it be better to enhance exiting plugin instead of writing custom plugin; 
the advantage is
1.  every body else can use it
2. we don't have to manage the custom plugin 

thanks
